### PR TITLE
Task/scan test reorg

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR reorganizes the scan functional test files to match the organization of other functional tests.
- Also, using execution policies and camp resources from forall functional tests

### CODE REVIEW

File: test/functional/forall/CMakeLists.txt [hunk 1]
```diff
@@ -6,5 +6,7 @@
 ###############################################################################
 
 add_subdirectory(indexset)
+add_subdirectory(indexset-view)
```
Comment (davidbeckingsale) [inline test/functional/forall/CMakeLists.txt:9]: Looks like these are added here and in #793 - is that correct?

Comment (rhornung67) [reply test/functional/forall/CMakeLists.txt:9]: Yup. I put up three PRs yesterday, each including the previous commits.  Probably not a great idea. I was trying to see if I could get the travis intel build to not timeout.  My last PR culls out a bunch of old tests that are covered in the new organization.

Comment (davidbeckingsale) [reply test/functional/forall/CMakeLists.txt:9]: That's fine, one good way to handle that kind of PR dependency is to have them merge into each other (e.g. set the target branch of each PR to the preceeding one) and only have the first one target develop.

Comment (rhornung67) [reply test/functional/forall/CMakeLists.txt:9]: I thought about that, but too late. I'll change the base branches.

**Comment by willkill07** [comment]
@davidbeckingsale and @rhornung67 -- I snuck in some zero-length tests for scan.

**Comment by rhornung67** [comment]
Thanks @willkill07.

**Comment by rhornung67** [comment]
Thanks @willkill07.

**Comment by rhornung67** [comment]
@davidbeckingsale , @mdavis36 I merged my earlier PR. So I canged the base branch for this one back to develop. Please re-approve when you have a second. Thanks.

File: test/functional/scan/tests/test-scan-inclusive.hpp [hunk 2]
```diff
@@ -116,10 +119,13 @@ TYPED_TEST_P(ScanFunctionalTest, ScanInclusive)
 
 TYPED_TEST_P(ScanFunctionalTest, ScanInclusiveInplace)
 {
-  using EXEC_POLICY      = typename at<TypeParam, num<0>>::type;
-  using WORKING_RESOURCE = typename at<TypeParam, num<1>>::type;
-  using OP_TYPE          = typename at<TypeParam, num<2>>::type;
+  using EXEC_POLICY      = typename camp::at<TypeParam, camp::num<0>>::type;
+  using WORKING_RESOURCE = typename camp::at<TypeParam, camp::num<1>>::type;
+  using OP_TYPE          = typename camp::at<TypeParam, camp::num<2>>::type;
```
Comment (rchen20) [inline test/functional/scan/tests/test-scan-inclusive.hpp:124]: Specifying the full camp scope is probably a good idea. I'll make a note to change this in my own test PRs.

Comment (rhornung67) [reply test/functional/scan/tests/test-scan-inclusive.hpp:124]: Yes, it helps make clear what's part of RAJA and what is not.  Also, we're trying to minimize the use of using directives in header files.

**Comment by rhornung67** [comment]
@davidbeckingsale @mdavis36 @rchen20 I pushed a change to disable forcing recursive inlining with the intel compiler. This reduced the travis intel build to less than 30 minutes, giving us headroom for now.  Please reapprove when you have a sec. Thanks.

**Pull request merged**